pkg/testing/gadgetrunner/gadgetrunner.go

```diff
@@ -215,7 +215,40 @@ func (g *GadgetRunner[T]) RunGadget() {
 	require.NoError(g.testCtx, err, "running gadget error")
 }
 
+const (
+	defaultDomain      = "ghcr.io"
+	officialRepoPrefix = "inspektor-gadget/gadget/"
+	// localhost is treated as a special value for domain-name. Any other
+	// domain-name without a "." or a ":port" are considered a path component.
+	localhost = "localhost"
+)
+
+// splitIGDomain splits a repository name to domain and remote-name.
+// If no valid domain is found, the default domain is used. Repository name
+// needs to be already validated before.
+// Inspired on https://github.com/distribution/reference/blob/v0.5.0/normalize.go#L126
+// TODO: Ideally we should use the upstream function but docker.io is hardcoded there
+// https://github.com/distribution/reference/blob/v0.5.0/normalize.go#L31
+func splitIGDomain(name string) (domain, remainder string) {
```

Comment:
I guess there was a reason to copy the function instead of using https://github.com/inspektor-gadget/inspektor-gadget/blob/004c8916cb6e1928d567bff1ae3551ecbf3a9559/pkg/oci/oci.go#L652

What is the reason?